### ISSUE TITLE
'is mapping' work around?

### ISSUE THREAD

**Issue by zanhsieh** [saltstack-formulas/logrotate-formula#14]
Hi, I encounter this issue on salt 2015.5.0 @ CentOS 6:

```
[root@master /]# salt 'minion1' state.highstate saltenv=os_tb
minion1:
    Data failed to compile:
----------
    No matching sls found for 'logrotate.conf' in env 'os_tb'
----------
    Rendering SLS 'os_tb:logrotate.jobs' failed: Jinja syntax error: no test named 'mapping'; line 22

---
[...]
    - require:
      - pkg: logrotate
    - watch_in:
      - service: {{ logrotate.service }}
    - context:
      {% if value is mapping %}    <======================
      path: {{ value.get('path', key) }}
      data: {{ value.get('config', []) }}
      {% else %}
      path: {{ key }}
      data: {{ value }}
[...]

---
[root@master /]# salt-master --versions
           Salt: 2015.5.0
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
```

Besides upgrade jinja2 per [this link](https://github.com/saltstack/salt/issues/11267), just wonder if there's anyway to go around. Thanks.

**Comment by zanhsieh** [comment]
My other configuration in case it's needed:

```
/srv/salt/os/tb/top.sls
os_tb:
  '*':
    - logrotate.conf
    - logrotate.jobs

/srv/pillar/os/tb/top.sls
os_tb:
  '*':
    - logrotate

/srv/pillar/os/tb/logrotate.sls
# vim: sts=2 ts=2 sw=2 et ai
logrotate:
  lookup:
    pkg: logrotate
    service: crond
  jobs:
    yum:
      path: /var/log/yum.log
      config:
        - missingok
        - notifempty
        - yearly
        - create 0600 root root
```

**Comment by iggy** [comment]
Your best bet is to upgrade jinja probably. You might be able to roll back to a version before that was introduced (try git checkout 8f64b9ab0333c5d6fe934e94b2e31108a7a17ec6 )

**Comment by zanhsieh** [comment]
Problem indeed solved after rollback to 8f64b9a. I will close this issue.

**Comment by grizo** [comment]
@zanhsieh rather than rolling back, i ended up updating Jinja2 via pip then restarting the minion on el6 nodes. just incase anyone else runs into this issue.
